### PR TITLE
[docs] Release notes for 2.23.1.0

### PR DESCRIPTION
**Pull request by aishwarya24**
@netlify /preview/releases/ybdb-releases/v2.23/#v2.23.1.0

### CODE REVIEW

File: docs/content/preview/releases/yba-releases/v2.23.md [hunk 1]
```diff
@@ -15,6 +15,269 @@ What follows are the release notes for all releases in the YugabyteDB Anywhere (
 
 For an RSS feed of all release series, point your feed reader to the [RSS feed for releases](../index.xml).
 
+## v2.23.1.0 - September 13, 2024 {#v2.23.1.0}
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.23.md:18]:
```suggestion
## v2.23.1.0 - November 1, 2024 {#v2.23.1.0}
```

File: docs/content/preview/releases/yba-releases/v2.23.md [hunk 2]
```diff
@@ -15,6 +15,269 @@ What follows are the release notes for all releases in the YugabyteDB Anywhere (
...
+
+### Change log
+
+<details>
+  <summary>View the detailed changelog</summary>
+
+### Improvements
+
+* Allows adding an extra imagePullSecret to the StatefulSet in Kubernetes-based YBA 2.20.x installations. PLAT-13465
+* Enables the addition of an extra imagePullSecret to the StatefulSet in Kubernetes based installations. PLAT-13465
```
Comment (ddhodge) [inline docs/content/preview/releases/yba-releases/v2.23.md:41]:
```suggestion
```

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 3]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Introduces the `yb_make_next_ddl_statement_nonincrementing` configuration parameter to reduce catalog version increments during many consecutive DDLs, lowering memory and latency spikes. {{<issue 23786>}}
+* Updates advisory lock error message with a workaround hint instead of `not yet implemented.` {{<issue 23881>}}
+* Allows certain ALTER VIEW commands in the parser without requiring additional changes. {{<issue 23117>}}
+* Includes a modification to ysql_dump and Postgres that ensures all pg_class and pg_type OID preservation across a backup/restore when passing the include-yb-metadata flag. {{<issue 23304>}}
+* Streamlines sys table prefetching by automatically resetting catalog read time. {{<issue 23502>}}
+* Simplifies several ybc_pggate functions for better value return and removes unused functions. {{<issue 23513>}}
+* Revamps the restoration of old backups by only applying the check for `pg_class OID` if the new YSQL configuration parameter, `yb_ignore_heap_pg_class_oids`, is enabled. {{<issue 23304>}}
+* Removes unused `connected_database_` field from PgSession class for code cleanliness. {{<issue 23608>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:69]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 4]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+#### DocDB
+
+* Reduces risk of WAL over-garbage collection in xCluster and avoids unnecessary flushes by modifying the WAL retention policy. {{<issue 22862>}}
+* Enables reduction of duplicate code and custom flag filtering through `GetFlagInfos` relocation. {{<issue 23632>}}
+* Refactors in-memory TSDescriptors to use SysServerEntryPB for further application towards persistent TSDescriptors in the system catalog. {{<issue 22925>}}
+* Accelerates xCluster setup for tables by reducing setup time, storing all intermediate state in-memory, and limiting writes to `sys_catalog`. {{<issue 23286>}}
+* Allows adding new CatalogEntries without modifying the sys-catalog-tool through using helper functions. {{<issue 23235>}}
+* Refactors method on main table scan for abstract sourcing of point-lookup keys, enabling keys from secondary vector index. {{<issue 23374>}}
+* Enables reading stored vector index data from DocDB. {{<issue 23460>}}
+* Removes unnecessary commented code for a cleaner and more efficient codebase. {{<issue 18407>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:94]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 5]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Updates YCQL tables `system_schema.functions` and `system_schema.aggregates` to align with Cassandra, aiding Cortex migration. {{<issue 3893>}}
+
+#### DocDB
+
+* Reduces risk of WAL over-garbage collection in xCluster and avoids unnecessary flushes by modifying the WAL retention policy. {{<issue 22862>}}
+* Enables reduction of duplicate code and custom flag filtering through `GetFlagInfos` relocation. {{<issue 23632>}}
+* Refactors in-memory TSDescriptors to use SysServerEntryPB for further application towards persistent TSDescriptors in the system catalog. {{<issue 22925>}}
+* Accelerates xCluster setup for tables by reducing setup time, storing all intermediate state in-memory, and limiting writes to `sys_catalog`. {{<issue 23286>}}
+* Allows adding new CatalogEntries without modifying the sys-catalog-tool through using helper functions. {{<issue 23235>}}
+* Refactors method on main table scan for abstract sourcing of point-lookup keys, enabling keys from secondary vector index. {{<issue 23374>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:92]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 6]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+
+#### YCQL
+
+* Updates YCQL tables `system_schema.functions` and `system_schema.aggregates` to align with Cassandra, aiding Cortex migration. {{<issue 3893>}}
+
+#### DocDB
+
+* Reduces risk of WAL over-garbage collection in xCluster and avoids unnecessary flushes by modifying the WAL retention policy. {{<issue 22862>}}
+* Enables reduction of duplicate code and custom flag filtering through `GetFlagInfos` relocation. {{<issue 23632>}}
+* Refactors in-memory TSDescriptors to use SysServerEntryPB for further application towards persistent TSDescriptors in the system catalog. {{<issue 22925>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:89]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 7]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Refines the PatchStatus function by storing relation id directly in the relations vector, avoiding the need to call the PgSession::LoadTable method. {{<issue 24040>}}
+
+#### YCQL
+
+* Updates YCQL tables `system_schema.functions` and `system_schema.aggregates` to align with Cassandra, aiding Cortex migration. {{<issue 3893>}}
+
+#### DocDB
+
+* Reduces risk of WAL over-garbage collection in xCluster and avoids unnecessary flushes by modifying the WAL retention policy. {{<issue 22862>}}
+* Enables reduction of duplicate code and custom flag filtering through `GetFlagInfos` relocation. {{<issue 23632>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:88]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 8]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Removes unnecessary commented code for a cleaner and more efficient codebase. {{<issue 18407>}}
+* Enables scanning of the `sequences_data` table for xCluster replication, aiding in the bootstrapping process by integrating current sequence information into the WAL from the source universe. {{<issue 23493>}}
+* Introduces `DEFINE_NEW_INSTALL_VALUE` macro to enable controlled rollout of new features in fresh clusters. {{<issue 23505>}}
+* Enables taking into account of changed data during a vector index data update. {{<issue 23524>}}
+* Adds utilities for HNSW vector index implementation, benchmarking, and file reading. {{<issue 23376>}}
+* Removes the `CallVisitor` template function to simplify the sys-catalog-tool code. {{<issue 23548>}}
+* Allows each sequence to have an update in the WAL for seamless xCluster replication. {{<issue 23493>}}
+* Introduces a command-line tool for building, validating, and benchmarking HNSW indexes with Usearch implementation. {{<issue 23556>}}
+* Offers support for different vector coordinate types including vector-related templates. {{<issue 23613>}}
+* Imports hnswlib header-only library for enhancements. {{<issue 23609>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:103]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 9]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Introduces a new gflag `max_disk_throughput_mbps` for automated control of write rejections when disk is full, replacing `reject_writes_min_disk_space_aggressive_check_mb`. {{<issue 23373>}}
+* Allows easier debugging with additional logging and updated log prefixes. {{<issue 23702>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:105]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 10]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Ensures that ddl_queue and replicated_ddls do not colocate on the same tablet in colocated databases. {{<issue 23728>}}
+* Integrates hnswlib into the vector indexing framework, adding wrapper and allowing different distance calculations. {{<issue 23752>}}
+* Eliminates the capability to rename xCluster replication groups, reducing potential confusion. {{<issue 23778>}}
+* Enables sequences replication feature in source universe by creating `sequences_data` table and incorporating an outgoing stream. {{<issue 23917>}}
+* Enables shared memory by default for executing read and write pg client queries to enhance performance. {{<issue 23820>}}
+* Streamlines automatic syncing of inline third-party dependencies and breaks down `thirdparty_tool` into separate modules. {{<issue 23846>}}
+* Enables automatic application of DDL changes in xCluster using `automatic_ddl_mode` proto field. {{<issue 23860>}}
+* Enables display of cluster_uuid in tserver /varz endpoint for better test and call_home tracking. {{<issue 23864>}}
+* Unutilized non-transactional snapshot code gets removed for a cleaner transactional snapshot flow. {{<issue 23428>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:114]: Lets remove this.

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 11]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Ensures correct functioning of `search_path` by skipping its value's enquote during transaction setup. {{<issue 23440>}}
+* Enhances sequence generation consistency by making backend process stick to a logical connection when initializing a sequence variable with `nextval` function. {{<issue 23601>}}
+* Ensures DDL operations success with a sleep interval after database catalog version updates when Connection Manager is active. {{<issue 23756>}}
+* Updates the callhome URL to use HTTPS for enhanced security. {{<issue 23947>}}
+* Includes PID in yb_active_session_history for easy correlation with actual processes in YSQL, YCQL, and background activities. {{<issue 23070>}}
+* Allows reduction of thread stack size to avoid Linux hugepage backing. {{<issue 23927>}}
+* Allows explicit addition of gFlags to `gflag_allowlist.txt` for secure callhome data collection, plus `version_info` in tserver data. {{<issue 24103>}}
+* Adds a metric to alert users when the master follower heartbeat delay is too high. {{<issue 21178>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:123]:
```suggestion
* Adds `max_follower_heartbeat_delay` metric to alert users when the master follower heartbeat delay is too high. {{<issue 21178>}}
```

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 12]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Removes the need to set `set_cdc_min_replicated_index` during xCluster bootstrap for improved log retention. {{<issue 24105>}}
+* Ensures the user interface accurately displays disk size, even when multiple data directories are used. {{<issue 23810>}}
+
+#### CDC
+
+* Allows dynamic table addition alongside table removal from CDC streams without needing to disable or restart. {{<issue 23581>}}
+* Adds a yb-admin option to disable dynamic table addition when creating new streams. {{<issue 23482>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:130]:
```suggestion
* Adds `DYNAMIC_TABLES_DISABLED` yb-admin option to disable dynamic table addition when creating new streams. {{<issue 23482>}}
```

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 13]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* Adds HELP and TYPE metadata to YSQL Prometheus metrics endpoint for better metrics accessibility. {{<issue 23578>}}
+* Corrects an issue causing heap-buffer-overflow errors during ysqlconnmgr ASAN tests. {{<issue 23913>}}
+* Fixes a crash caused by using the `pg_hint_plan` with the `hint_table` enabled. {{<issue 23547>}}
+* Updates the copyright string from `YugaByteDB` to the correct version `YugabyteDB`, ensuring error-free linter runs. {{<issue 23940>}}
+* Allows for a modified calculation of net connections in YSQL connection manager with the warmup random mode. {{<issue 23588>}}
+* Prevents background parallel workers from executing check/assign hooks of YSQL configuration parameters, eliminating potential crashes. {{<issue 23787>}}
+* Modifies backup/restore process to skip column name checks for indexes, allowing for successful restoration even with renamed columns. {{<issue 24207>}}
+* Fixes crashes caused by using a RowComparisonExpression on a reordered primary key index. {{<issue 23824>}}
+* Replaces deprecated function `shared_ptr::unique` to resolve compilation errors on macOS. {{<issue 24012>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:182]: Lets remove this

File: docs/content/preview/releases/ybdb-releases/v2.23.md [hunk 14]
```diff
@@ -15,6 +15,235 @@ What follows are the release notes for the YugabyteDB v2.23 release series. Cont
...
+* The update stabilizes WriteBuffer, preventing potential crashes when constructed over non-zeroed memory. {{<issue 23960>}}
+* Adds a null check to prevent a system crash when cleaning up a recently applied transaction state after a flush operation. {{<issue 24026>}}
+* Resolves the shutdown issue with the shared exchange causing TServer to hang by properly managing concurrent threads and session destruction, and sets `pg_client_use_shared_memory` to false by default on Mac to enhance performance. {{<issue 24000>}}
+* Ensures "create table" requests for colocated tables don't fail due to mistaken tablet limit checks. {{<issue 23922>}}
+
+#### CDC
+
+* Refactors CDCSDK to prevent data loss during workload with single shard transactions by ensuring WAL messages are committed before being replicated. {{<issue 19294>}}
+* Reduces disk IO overhead by optimizing the tablet-meta update process in CDCService. {{<issue 22805>}}
+* Introduces three new yb-admin commands for managing user tables in a CDC stream and addresses a race condition issue by removing an optimization in the `UpdateCheckpointForTabletEntriesInCDCState` method. {{<issue 22876>}},{{<issue 22835>}},{{<issue 22773>}}
```
Comment (premkumr) [inline docs/content/preview/releases/ybdb-releases/v2.23.md:233]:
```suggestion
* Introduces three new yb-admin commands : `disable_dynamic_table_addition_in_change_data_stream`, 
  `remove_user_table_from_change_data_stream`, 
 `validate_and_sync_cdc_state_table_entries_on_change_data_stream` for managing user tables in a CDC stream and addresses a race condition issue  {{<issue 22876>}},{{<issue 22835>}},{{<issue 22773>}}
```

**Review by premkumr — APPROVED** [review]
From next time, I'll review the sheet .

**Pull request merged**